### PR TITLE
Update Developer Guide

### PR DESCRIPTION
**Pull request by yanghengtang**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -1217,30 +1219,213 @@ testers are expected to do more *exploratory* testing.
...
 1. Deleting a patient while all patients are being shown
 
    1. Prerequisites: List all patients using the `list_p` command. Multiple patients in the list.
 
-   1. Test case: `delete_p 1`<br>
+   2. Test case: `delete_p 1`<br>
       Expected: First patient is deleted from the list. Details of the deleted patient shown in the status message.
 
-   1. Test case: `delete_p 0`<br>
+   3. Test case: `delete_p 0`<br>
```
Comment (derekjxtan) [inline docs/DeveloperGuide.md:1357]: Indicate invalid index?

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -1217,30 +1219,213 @@ testers are expected to do more *exploratory* testing.
...
       Expected: No patient is deleted. Error details shown in the status message.
 
-   1. Other incorrect delete commands to try: `delete_p`, `delete_p x`, `...` (where x is larger than the list size)<br>
+   4. Other incorrect delete commands to try: `delete_p`, `delete_p x`, `...` (where x is larger than the list size)<br>
```
Comment (derekjxtan) [inline docs/DeveloperGuide.md:1360]: remove `...`? not sure what its supposed to mean now that i think about it

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -1217,30 +1219,213 @@ testers are expected to do more *exploratory* testing.
...
       Expected: Similar to previous.
 
+### Viewing a patient
+
+1. Viewing a patient while all patients are being shown
+
+    1. Prerequisites: List all patients using the `list_p` command. Multiple patients in the list.
+
+    2. Test case: `view_p 0`<br>
```
Comment (derekjxtan) [inline docs/DeveloperGuide.md:1369]: indicate invalid index?

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -1217,30 +1219,213 @@ testers are expected to do more *exploratory* testing.
...
+       Expected: Similar to previous.
+
+    5. Test case (Multiple parameters): `add_tag_p 1 t\Depression t\Depression`<br>
+       Expected: Similar to previous.
+
+### Deleting a medical condition
+
+1. Deleting a medical condition from patient while all patients are being shown
+
+    1. Prerequisites: List all patients using the `list_p` command. Multiple patients in the patient list. Ensure the first patient have the medical condition "Depression".
```
Comment (derekjxtan) [inline docs/DeveloperGuide.md:1400]: `Ensure the first patient have` change to `ensure that the first patient has`

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -1217,30 +1219,213 @@ testers are expected to do more *exploratory* testing.
...
+   2. Re-launch the app by double-clicking the jar file.<br>
        Expected: The most recent window size and location is retained.
 
+### Adding a patient
+
+1. Adding a patient
+
+    1. Prerequisites: There exist no patient with NRIC `T1258979J` in the database.
+
+    2. Test case (Valid parameters): `add_p n\Bunny Cai Hong ic\T1258979J p\83425673`<br>
```
Comment (derekjxtan) [inline docs/DeveloperGuide.md:1237]: your siambu gf?

Comment (yanghengtang) [reply docs/DeveloperGuide.md:1237]: you like?

image

**Review by derekjxtan — APPROVED** [review]

**Pull request merged**